src/components/expression/expression-panel.js

```diff
@@ -224,16 +218,66 @@ genotet.ExpressionPanel.prototype.updateGenes = function(gene) {
       text: gene
     };
   });
-
   var profile = this.container.find('#profile select').empty();
-  this.selectProfiles_ = profile.select2({
-    data: genes,
-    multiple: true
-  });
-  this.container.find('#profile .select2-container').css({
-    width: '100%'
-  });
-  this.updateGeneSelections();
+
+  if (!genes.length) {
+    return;
+  }
+
+  $.fn.select2.amd.require(['select2/data/array', 'select2/utils'],
```

Comment:
Yes, above issues are solved except for the search matching issue.

On Mon, Apr 25, 2016 at 9:21 PM, Bowen Yu <notifications@github.com> wrote:

> In src/components/expression/expression-panel.js
> <https://github.com/ViDA-NYU/genotet/pull/151#discussion_r61016628>:
>
> >    var profile = this.container.find('#profile select').empty();
> > -  this.selectProfiles_ = profile.select2({
> > -    data: genes,
> > -    multiple: true
> > -  });
> > -  this.container.find('#profile .select2-container').css({
> > -    width: '100%'
> > -  });
> > -  this.updateGeneSelections();
> > +
> > +  if (!genes.length) {
> > +    return;
> > +  }
> > +
> > +  $.fn.select2.amd.require(['select2/data/array', 'select2/utils'],
>
> IIRC when you type-in any character you can only list those items matching
> the characters. That's equivalent to a search box.
>
> Has the above issue been resolved?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/151/files/90437f8ff78facbc443fa2717a721540a82c9157#r61016628>
>